src/Middleware/AddHttp2ServerPush.php

```diff
@@ -43,17 +43,20 @@ public function handle(Request $request, Closure $next)
      *
      * @return $this
      */
-    protected function generateAndAttachLinkHeaders(Response $response)
+    protected function generateAndAttachLinkHeaders(Response $response, $limit = null)
     {
         $headers = $this->fetchLinkableNodes($response)
             ->flatten(1)
             ->map(function ($url) {
                 return $this->buildLinkHeaderString($url);
-            })->filter()
-            ->implode(',');
+            })->filter();
 
-        if (!empty(trim($headers))) {
-            $this->addLinkHeader($response, $headers);
+        if($limit) $headers = $headers->take($limit);
```

Comment:
interestingly enough, if you look at what is being used under the hood on the `take()` method, it is `array_slice()`. If you pass `null` in as the limit, it will return the original collection or essentially will have no effect.

This eliminates the need for doing a conditional to check for the limit since we are setting it to `null` by default here.